gemrb/GUIScripts/iwd/CharGen.py

```diff
@@ -1524,17 +1524,21 @@ def SkillsPress():
 	LUSkillsSelection.SkillsNullify (MyChar)
 
 	if SkillsState == 0:
-		GemRB.SetVar ("HatedRace", 0)
-		if IsThief!="*":
-			SkillsSelect()
-		elif DruidSpell!="*":
+		if DruidSpell!="*":
 			Skill = GemRB.LoadTable("SKILLRNG").GetValue(str(Level), "STEALTH")
 			GemRB.SetPlayerStat (MyChar, IE_STEALTH, Skill)
-			RacialEnemySelect()
 		elif IsBard!="*":
 			Skill = GemRB.LoadTable(IsBard).GetValue(str(Level), "PICK_POCKETS")
 			GemRB.SetPlayerStat (MyChar, IE_PICKPOCKET, Skill)
-			SkillsState = 1
+
+		GemRB.SetVar ("HatedRace", 0)
+		GemRB.SetVar ("RacialEnemy", 0)
+		GemRB.SetVar ("RacialEnemyIndex", 0)
```

Comment:
Are these two used later as well? If so, sounds like a refactoring waiting to happen. Either way we should just save and reset the stat directly in RacialEnemySelect and kids.

Reply:
They are used in to display racial enemy in character description screen. Without resetting these, selecting ranger, choosing racial enemy, going back, and say, changing class to druid will print later race enemy.

> Either way we should just save and reset the stat directly in RacialEnemySelect and kids.

It has to be reset before it, because it's not guaranteed that `RacialEnemySelect` will be called.

Reply:
With direct storage, the description code can check against the stat as well. But a refactoring is best done as a separate PR.